connector/connect/server/src/main/scala/org/apache/spark/sql/connect/utils/ErrorUtils.scala

```diff
@@ -213,6 +221,30 @@ private[connect] object ErrorUtils extends Logging {
       .exists(_.toString.contains("org.apache.spark.sql.execution.python"))
   }
 
+  def handleNonQueryError[V](
+      opName: String,
+      observer: StreamObserver[V],
+      sessionHolderOpt: Option[SessionHolder],
+      callback: Option[() => Unit] = None): PartialFunction[Throwable, Unit] = {
```

Comment:
does this need to be a separate function from `handleError`, or could the "callback" be called into the previous function?
At a glance `handleError` seems to be a compatible superset of functionality.